cake/build-settings.cake

```diff
@@ -1,23 +1,354 @@
-#load ./ci.cake
-#load ./packaging.cake
-#load ./package-checks.cake
-#load ./test-results.cake
-#load ./package-tests.cake
-#load ./package-tester.cake
-#load ./header-check.cake
-#load ./local-tasks.cake
-
-public class BuildSettings
+public static class BuildSettings
 {
-    public BuildSettings(ISetupContext context)
-    {
+	private static BuildSystem _buildSystem;
+
+	public static void Initialize(
+      // Required parameters
+        ICakeContext context,
+        string title,
+        string githubRepository,
+
+      // Optional Parameters
+        bool suppressHeaderCheck = false,
+        string[] standardHeader = null,
+        string[] exemptFiles = null,
+        
+        string solutionFile = null,
+        string[] validConfigurations = null,
+        string githubOwner = "NUnit",
+
+		bool msbuildAllowPreviewVersion = false,
+		Verbosity msbuildVerbosity = Verbosity.Minimal,
+
+        string unitTests = null, // Defaults to "**/*.tests.dll|**/*.tests.exe" (case insensitive)
+        UnitTestRunner unitTestRunner = null, // If not set, NUnitLite is used
+        string unitTestArguments = null
+        )
+	{
+        // Required arguments
         if (context == null)
-            throw new System.ArgumentNullException(nameof(context));
+            throw new ArgumentNullException(nameof(context));
+		if (title == null)
+			throw new ArgumentNullException(nameof(title));
+		if (githubRepository == null)
+			throw new ArgumentNullException(nameof(githubRepository));
+
+        Context = context;
+        Title = title;
+        GitHubRepository = githubRepository;
+
+		// NOTE: Order of initialization can be sensitive. Obviously,
+		// we have to set any properties in this method before we
+		// make use of them. Less obviously, some of the classes we
+		// construct here have dependencies on certain properties
+		// being set before the constructor is called. I have
+		// tried to annotate such dependencies below.
+
+        _buildSystem = context.BuildSystem();
+	
+		// If not specified, uses TITLE.sln if it exists or uses solution
```

Comment:
This comment is an implementation detail of DeduceSolutionFile.  I would avoid the comment here and keep one on the helper method only in case the logic changes and this comment gets out of date.